dlpy/applications/efficientnet.py

```diff
@@ -0,0 +1,292 @@
+        Valid Values: 'none', 'random'
+
+
+    Returns
+    -------
+    :class:`Model`
+
+    References
+    ----------
+    https://arxiv.org/pdf/1905.11946.pdf
+
+    '''
+
+    def _make_divisible(v, divisor, min_value=None):
+        # make number of channel divisible
+        if min_value is None:
+            min_value = divisor
+        new_v = max(min_value, int(v + divisor / 2) // divisor * divisor)
+        # make sure that round down does not go down by more than 10%.
+        if new_v < 0.9 * v:
+            new_v += divisor
+        return new_v
+
+    def round_filters(filters, width_coefficient, depth_divisor):
+        # round number of filters based on width multiplier and depth_divisor
+
+        filters *= width_coefficient
+        new_filters = int(filters + depth_divisor / 2) // depth_divisor * depth_divisor
+        new_filters = max(depth_divisor, new_filters)
+        # Make sure that round down does not go down by more than 10%.
+        if new_filters < 0.9 * filters:
+            new_filters += depth_divisor
+        return int(new_filters)
+
+    def round_repeats(repeats, depth_coefficient):
+        # round number of repeats based on depth multiplier
+
+        return int(math.ceil(depth_coefficient * repeats))
+
+    def _MBConvBlock(inputs, in_channels, out_channels, ksize, stride, expansion, se_ratio, stage_id, block_id,
+                     noskip=False, activation_fn='relu'):
+        """
+        Inverted Residual Block
+
+        Parameters
+        ----------
+        inputs:
+            Input tensor
+        in_channels:
+            Specifies the number of input tensor's channel
+        expansion:
+            expansion factor always applied to the input size.
+        stride:
+            the strides of the convolution
+        alpha:
+            width multiplier.
+        filters:
+            the dimensionality of the output space.
+        block_id:
+            block id used for naming layers
+
+        """
+
+        # mobilenetv2 block is also known as inverted residual block, which consists of three convolutions:
+        # the first is 1*1 convolution for expansion
+        # the second is depthwise convolution
+        # the third is 1*1 convolution without any non-linearity for projection
+
+        x = inputs
+        prefix = 'stage_{}_block_{}'.format(stage_id, block_id)
+        n_groups = in_channels  # for expansion=1, n_groups might be different from pointwise_filters
+
+        if expansion > 1:
+            # For MobileNet V2, expansion>1 when stage>0
+            n_groups = int(expansion * in_channels)  ## update n_groups
+            x = Conv2d(n_groups, 1, include_bias=False, act='identity',
+                       name=prefix + 'expand')(x)
+            x = BN(name=prefix + 'expand_BN', act='identity')(x)
+
+        # Depthwise convolution with kernel size : 3 or 5
+        x = GroupConv2d(n_groups, n_groups, ksize, stride=stride, act='identity',
```

Comment:
if ksize takes only 3 or 5 then you need to specify this in the docstring - maybe some checks here

Reply:
The kernel size is specified from the block's parameters, which can be different from 3 and 5. So I updated the comment.